### DISCUSSION TITLE
In what situation does BasedMovement handling necessary for als camera component?

### DISCUSSION THREAD

**Discussion by ameaninglessname** [Sixze/ALS-Refactored#419] [Q&A] [answered]
I couldn't see a difference after hard-coded `bMovementBaseHasRelativeRotation` to false with `UCharacterMovementComponent::bIgnoreBaseRotation ` set to `false` when standing on `B_Als_RotatingObject` 🤔 (relative rotation is unchanged in both case)

the relative code starts here (so the code looks unnecessary):
https://github.com/Sixze/ALS-Refactored/blob/main/Source/ALSCamera/Private/AlsCameraComponent.cpp#L170-L171

**Comment by Sixze** [comment]
There is a difference, as the camera will start rotating along with `B_Als_RotatingObject`, thus keeping its relative rotation unchanged if `bIgnoreBaseRotation` is off.

**Reply by ameaninglessname** [reply]
Yes, you're right.
And sorry, there was some key omissions in my description, just updated.

**Reply by Sixze** [reply, marked as answer]
Press Shift+7 to enable camera debug shapes, and you'll see that in one case they follow the rotating character with some delay, and in the other case they don't, because all camera calculations are done relative to the rotating platform.

**Reply by ameaninglessname** [reply]
Thanks! ♥ I got that.
